src/HDF5.jl

```diff
@@ -323,6 +323,13 @@ cset(::Type{ASCIIChar}) = H5T_CSET_ASCII
 
 hdf5_type_id(::Type{C}) where {C<:CharType} = H5T_C_S1
 
+# global configuration for complex support
+complex_support = true
+complex_field_names = ("r", "i")
+enable_complex_support() = global complex_support = true
+disable_complex_support() = global complex_support = false
+set_complex_field_names(real::String, imag::String) = global complex_field_names = (real, imag)
+
```

Comment:
do we really need these global setter functions?

What's the rationale for being able to enable/disable complex support? Same with set_complex_field_names?

Reply:
In any case a better way to deal with such global constants is to use `Ref`'s with appropriate setters.

Reply:
I implemented it this way following what is done by h5py and for compatibility with existing practice/workflows. Since there is no actual hdf5 standard for complex numbers allowing the fieldnames to be set dynamically lets you interface more easily with existing conventions. For example I think that octave uses real/imag and I have also seen re/im. Also it's conceivable that somebody is storing a compound type with these fieldnames that doesn't represent a complex number. I'm not sure if this flexibility is necessary/worth it but I decided to follow the h5py precedent in allowing it.

Reply:
I have trouble seeing why we would ever need to disable complex support?
Surely it's not just there for performance?

Reply:
Not sure why they would want to do that, but even if they did couldn't they just change the field names?

Reply:
@kleinhenz that makes sense 

Can we please update all the global constants to use refs (it's  precompile friendly, as opposed to globals)

```julia
const COMPLEX_SUPPORT = Ref(true) 
const COMPLEX_FIELD_NAMES = Ref(("r", "i"))

enable_complex_support() = COMPLEX_SUPPORT[] = true
disable_complex_support() = COMPLEX_SUPPORT[] = false
set_complex_field_names(real::AbstractString, imag::AbstractString) =  COMPLEX_FIELD_NAMES[] = ((real, imag))
```


And update everywhere else to use `COMPLEX_SUPPORT[]` instead of  `complex_support`

Reply:
I can see the rational of being able to disable this feature. It allows to to read the complex numbers in its "raw" representation, which otherwise would not be possible. It also does not really hurt making this optional.

Reply:
Looking at `h5py` they don't offer a seperate way to enable disable support.

I kinda agree with @benchislett that even if they would want to do that they could just change the field names. 

I'm not 100% convinced which approach is best. I'm leaning towards  no option to enable disable support since that seems a little overboard. OTOH we can always remove this non-user facing functions in the future if deemed overkill.

Reply:
I think you are correct that h5py doesn't provide a mechanism for completely disabling complex support. I don't have a strong preference on this either. I lean towards including it just to be the least disruptive to existing workflows but I could see it the other way also. 

One point in favor is that changing the field names will prevent you from reading data as complex when you don't want to but it can't prevent you from accidentally writing complex data in an unwanted format where previously it would have thrown an exception so I think including the enable/disable thing is somewhat safer for people with a preexisting way of handling complex data.

Reply:
Ok that's convincing enough 👍